homeassistant/components/traccar/device_tracker.py

```diff
@@ -109,26 +110,38 @@ def __init__(self, api, hass, async_see, scan_interval,
         self._scan_interval = scan_interval
         self._async_see = async_see
         self._api = api
+        self.connected = None
         self._hass = hass
 
     async def async_init(self):
         """Further initialize connection to Traccar."""
-        await self._api.test_connection()
-        if self._api.authenticated:
-            await self._async_update()
-            async_track_time_interval(self._hass,
-                                      self._async_update,
-                                      self._scan_interval)
-
-        return self._api.authenticated
+        await self._async_update()
+        async_track_time_interval(self._hass,
+                                  self._async_update,
+                                  self._scan_interval)
 
     async def _async_update(self, now=None):
         """Update info from Traccar."""
-        _LOGGER.debug('Updating device data.')
-        await self._api.get_device_info(self._custom_attributes)
-        self._hass.async_create_task(self.import_device_data())
-        if self._event_types:
-            self._hass.async_create_task(self.import_events())
+        if self.connected is None:
+            # We should only get here on the first run.
+            await self._api.test_connection()
+            self.connected = self._api.connected
+            if not self._api.connected:
+                _LOGGER.error("Not connected to Traccar.")
+
+        if not self.connected:
```

Comment:
Why do we need a special case here? It could be merged with the case above.

Reply:
Merged some of it, is it enough?
I want to limit the amount of logs if the Traccar service is restarting (temporarily unavailable)